Can't compile against Python 3.7

I'm trying to compile pdal 1.7.2 against Python 3.7 but they seem to have changed their function signature a bit:

```
/build/pdal/src/PDAL-1.7.2-src/plugins/python/plang/Environment.cpp: In function ‘std::__cxx11::string pdal::plang::getTraceback()’:
/build/pdal/src/PDAL-1.7.2-src/plugins/python/plang/Environment.cpp:193:46: error: invalid conversion from ‘const char*’ to ‘char*’ [-fpermissive]
             char *d = PyUnicode_AsUTF8AndSize(r, &size);
                       ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
/build/pdal/src/PDAL-1.7.2-src/plugins/python/plang/Environment.cpp:211:42: error: invalid conversion from ‘const char*’ to ‘char*’ [-fpermissive]
         char *d = PyUnicode_AsUTF8AndSize(r, &size);
                   ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
/build/pdal/src/PDAL-1.7.2-src/plugins/python/plang/Environment.cpp: In function ‘std::__cxx11::string pdal::plang::readPythonString(PyObject*, const string&)’:
/build/pdal/src/PDAL-1.7.2-src/plugins/python/plang/Environment.cpp:269:38: error: invalid conversion from ‘const char*’ to ‘char*’ [-fpermissive]
     char *d = PyUnicode_AsUTF8AndSize(r, &size);
               ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
/build/pdal/src/PDAL-1.7.2-src/plugins/python/plang/Environment.cpp: In function ‘std::__cxx11::string pdal::plang::getTraceback()’:
/build/pdal/src/PDAL-1.7.2-src/plugins/python/plang/Environment.cpp:193:46: error: invalid conversion from ‘const char*’ to ‘char*’ [-fpermissive]
             char *d = PyUnicode_AsUTF8AndSize(r, &size);
                       ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
/build/pdal/src/PDAL-1.7.2-src/plugins/python/plang/Environment.cpp:211:42: error: invalid conversion from ‘const char*’ to ‘char*’ [-fpermissive]
         char *d = PyUnicode_AsUTF8AndSize(r, &size);
                   ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
/build/pdal/src/PDAL-1.7.2-src/plugins/python/plang/Environment.cpp: In function ‘std::__cxx11::string pdal::plang::readPythonString(PyObject*, const string&)’:
/build/pdal/src/PDAL-1.7.2-src/plugins/python/plang/Environment.cpp:269:38: error: invalid conversion from ‘const char*’ to ‘char*’ [-fpermissive]
     char *d = PyUnicode_AsUTF8AndSize(r, &size);
               ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
make[2]: *** [plugins/python/io/CMakeFiles/pdal_plugin_reader_numpy.dir/build.make:76: plugins/python/io/CMakeFiles/pdal_plugin_reader_numpy.dir/__/plang/Environment.cpp.o] Error 1
[ 52%] Linking CXX executable ../../bin/pdal_io_bpf_base_test
make[1]: *** [CMakeFiles/Makefile2:661: plugins/python/io/CMakeFiles/pdal_plugin_reader_numpy.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 52%] Building CXX object plugins/python/filters/CMakeFiles/pdal_plugin_filter_python.dir/PythonFilter.cpp.o
[ 53%] Building CXX object test/unit/CMakeFiles/pdal_support_test.dir/SupportTest.cpp.o
[ 54%] Building CXX object test/unit/CMakeFiles/pdal_stage_factory_test.dir/StageFactoryTest.cpp.o
make[2]: *** [plugins/python/filters/CMakeFiles/pdal_plugin_filter_python.dir/build.make:76: plugins/python/filters/CMakeFiles/pdal_plugin_filter_python.dir/__/plang/Environment.cpp.o] Error 1
```

Compiler is gcc 8.1.

--- a/plugins/python/plang/Environment.cpp
+++ b/plugins/python/plang/Environment.cpp
@@ -190,9 +190,9 @@ std::string getTraceback()
                 throw pdal::pdal_error("unable to get repr in getTraceback");
 #if PY_MAJOR_VERSION >= 3
             Py_ssize_t size;
-            char *d = PyUnicode_AsUTF8AndSize(r, &size);
+            const char *d = PyUnicode_AsUTF8AndSize(r, &size);
 #else
-            char *d = PyString_AsString(r);
+            const char *d = PyString_AsString(r);
 #endif
             mssg << d;
         }
@@ -208,9 +208,9 @@ std::string getTraceback()
             throw pdal::pdal_error("couldn't make string representation of traceback value");
 #if PY_MAJOR_VERSION >= 3
         Py_ssize_t size;
-        char *d = PyUnicode_AsUTF8AndSize(r, &size);
+        const char *d = PyUnicode_AsUTF8AndSize(r, &size);
 #else
-        char *d = PyString_AsString(r);
+        const char *d = PyString_AsString(r);
 #endif
         mssg << d;
     }
@@ -266,9 +266,9 @@ std::string readPythonString(PyObject* dict, const std::string& key)
         throw pdal::pdal_error("unable to get repr in readPythonString");
 #if PY_MAJOR_VERSION >= 3
     Py_ssize_t size;
-    char *d = PyUnicode_AsUTF8AndSize(r, &size);
+    const char *d = PyUnicode_AsUTF8AndSize(r, &size);
 #else
-    char *d = PyString_AsString(r);
+    const char *d = PyString_AsString(r);
 #endif
     ss << d;
 

